parquet-column/src/main/java/org/apache/parquet/column/values/factory/DefaultV2ValuesWriterFactory.java

```diff
@@ -84,10 +92,25 @@ private ValuesWriter getBooleanValuesWriter() {
   }
 
   private ValuesWriter getFixedLenByteArrayValuesWriter(ColumnDescriptor path) {
-    ValuesWriter fallbackWriter = new DeltaByteArrayWriter(
-        parquetProperties.getInitialSlabSize(),
-        parquetProperties.getPageSizeThreshold(),
-        parquetProperties.getAllocator());
+    final ValuesWriter fallbackWriter;
+    // Heuristic: enable BYTE_STREAM_SPLIT for DECIMAL and FLOAT16 columns.
+    final boolean useByteStreamSplit = parquetProperties.isExtendedByteStreamSplitEnabled()
```

Comment:
Why the different heuristics for V1 and V2?

Do we know if `BYTE_STREAM_SPLIT` would be better for `DECIMAL`/`FLOAT16` values than `DELTA_BYTE_ARRAY`? It would be nice to come up with some smart heuristics instead of having the client set it, but if we don't know which one is better in certain cases, I would vote on not having any heuristics.

Reply:
* Footprint-wise:
  1. `DELTA_BYTE_ARRAY` is, at least conceptually, suboptimal for FLBA as it will store both prefix and suffix lengths, even though one can be trivially deduced from the other.
  2. `DELTA_BYTE_ARRAY` is most certainly a bad fit for very short data such as `FLOAT16`.
  3. While https://issues.apache.org/jira/browse/PARQUET-2414 did exercise `BYTE_STREAM_SPLIT` on `DECIMAL` and `FLOAT16` values, we don't know if it can be beneficial or detrimental for other kinds of `FIXED_LEN_BYTE_ARRAY` column.

* CPU-wise, `BYTE_STREAM_SPLIT` should be much faster than `DELTA_BYTE_ARRAY`, though concrete benchmarks would be needed for parquet-mr (we do have benchmarks for Parquet C++).

That said, perhaps we should indeed not add any heuristics and instead let the user configure things column-by-column (combining heuristics + column selection would be a bit delicate, though).

Reply:
Maybe let the user decide for now and create an additional jira to implement smart heuristics. Then, some additional time can be spent on experimenting.